userspace/libscap/scap.h

```diff
@@ -1002,6 +1021,31 @@ uint64_t scap_get_driver_api_version(scap_t* handle);
  */
 uint64_t scap_get_driver_schema_version(scap_t* handle);
 
+
+/**
+ * Get hostname.
+ *
+ * Supports env variable lookup for cloud native integrations
+ * where the hostname can be equivalent to the Kubernetes pod name.
+ * Customizable over cmake setup SCAP_HOSTNAME_ENV_VAR.
+ */
+void scap_gethostname(scap_t* handle);
+
+/**
+ * Retrieve agent info.
+ */
+void scap_retrieve_agent_info(scap_t* handle);
+
+/**
+ * Retrieve machine info.
+ */
+void scap_retrieve_machine_info(scap_t* handle, uint64_t boot_time);
+
+/**
+ * Check is kernel.bpf_stats_enabled is set.
+ */
+void scap_get_bpf_stats_enabled(scap_t* handle);
```

Comment:
As @jasondellaluce correctly said these seem more like internal scap APIs rather than public ones, probably they should live in `scap-int.h` and we could have a dedicated `.c` file for the implementation (similar to what we do with `scap_userlists.c`

Reply:
@jasondellaluce and @Andreagit97 will do.

Reply:
oops almost forgot about the request to create a dedicated `.c` file, just pushed it, see last commit @Andreagit97.